xstate/action.py

```diff
@@ -8,10 +8,17 @@ def not_implemented():
 class Action:
     type: str
     exec: Callable[[], None]
+    data: Dict[str, Any]
 
-    def __init__(self, type: str, exec: Optional[Callable[[], None]] = not_implemented):
+    def __init__(
+        self,
+        type: str,
+        exec: Optional[Callable[[], None]] = not_implemented,
+        data: Dict[str, Any] = {},
```

Comment:
Added `data` because an action can contain data, like `{ "type": "log", "message": "whatever" }`, that `"message"` etc. is data that we need.